core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -195,12 +218,9 @@ public String toString() {
     return name();
   }
 
-  abstract String metadataLocation();
```

Comment:
Is the plan to use `operations()` on metadata tables to derive metadata location? wondering if we want to keep this method for backward compatibility/avoid breaking consumer

Reply:
This was package-private so it should be okay to remove it. I think using `HasTableOperations` is a better way to access the metadata location.

Reply:
I don't think `BaseMetadataTable` is public and this method was added for serialization. I am going to rework table serialization in the next PR and we won't depend on this method.

That said, I don't mind keeping this too if there is a good use case. I just thought it is a pretty internal API.

Reply:
makes sense, didn't notice that it was package-private so probably don't need to add back. Thanks for the explanation!